samanthas_telegram_bot/api_queries/auxil/data_maker.py

```diff
@@ -0,0 +1,111 @@
+import json
+
+from telegram import Update
+
+from samanthas_telegram_bot.api_queries.auxil.constants import DataDict
+from samanthas_telegram_bot.data_structures.context_types import UserData
+
+
+class DataMaker:
+    """Class for creating data dictionaries to be sent to backend."""
+
+    STATUS_AT_CREATION_STUDENT_TEACHER = "awaiting_offer"
+    STATUS_AT_CREATION_TEACHER_UNDER_18 = "active"
+
+    @staticmethod
+    def personal_info(user_data: UserData) -> DataDict:
+        return {
+            "communication_language_mode": user_data.communication_language_in_class,
+            "first_name": user_data.first_name,
+            "last_name": user_data.last_name,
+            "telegram_username": user_data.tg_username or "",
+            "email": user_data.email,
+            "phone": user_data.phone_number,
+            "utc_timedelta": f"{user_data.utc_offset_hour:02d}:{user_data.utc_offset_minute}:00",
+            "information_source": user_data.source,
+            "registration_telegram_bot_chat_id": user_data.chat_id,
+            "registration_telegram_bot_language": user_data.locale,
+        }
+
+    @classmethod
+    def student(cls, update: Update, personal_info_id: int, user_data: UserData) -> DataDict:
+        data: DataDict = {
+            "personal_info": personal_info_id,
+            "comment": user_data.comment,
+            "status": cls.STATUS_AT_CREATION_STUDENT_TEACHER,
+            "status_since": cls._format_status_since(update),
+            "can_read_in_english": user_data.student_can_read_in_english,
+            "is_member_of_speaking_club": False,  # TODO can backend set to False by default?
+            "age_range": user_data.student_age_range_id,
+            "availability_slots": user_data.day_and_time_slot_ids,
+            "non_teaching_help_required": user_data.non_teaching_help_types,
+            "teaching_languages_and_levels": user_data.language_and_level_ids,
+        }
+        if user_data.student_smalltalk_result:
+            data["smalltalk_test_result"] = json.dumps(
```

Comment:
I'm not sure, why you keep bytes in original_json. I will explain in the comment in the smalltalk module

Reply:
Fixed